server/functions/framework/cast.go

```diff
@@ -255,7 +255,7 @@ func getCast(mutex *sync.RWMutex,
 	}
 	// If there isn't a direct mapping, then we need to check if the types are array variants.
 	// As long as the base types are convertable, the array variants are also convertable.
-	if fromArrayType, ok := fromType.IsBaseIDArrayType(); ok {
+	if fromArrayType, ok := fromType.IsBaseIDArrayType(); ok && fromType != pgtypes.DoltgresTypeBaseID_Unknown {
```

Comment:
I don't think this addition is actually necessary. Since `unknown` will never be a target for a cast, it will never end up matching with anything. Was there a reason this was added? That may point to an issue or error elsewhere.

Reply:
One of the tests that fails is https://github.com/dolthub/doltgresql/blob/main/testing/go/types_test.go#L474 
The `unknown` type is not the target type, it's the `fromType` in this test trying to cast string `"{abcdefghij,NULL}"` to `CHARACTER VARYING[]` type. Since the `unknown` type qualifies as `array type`, it passes this check and panics trying to cast `string` to `[]any` in `oldVals := vals.([]any)`

Reply:
Ohh I see. I wonder if `unknown` should qualify as an array type then. I don't see anything in Postgres stating one way or another, so this is something to investigate for later. For now, we can leave a TODO here saying to look into it.

Reply:
IIRC, `unknown` was made an array type so that it would be valid to return in the case of errors on array types, but that may not be the valid thing to do. We use `AnyArray` as the error type in the `ARRAY` expression, but I'm also not sure if _that_ is the correct thing to do lol.